ceno_zkvm/src/tables/range/range_circuit.rs

```diff
@@ -116,3 +119,55 @@ impl<E: ExtensionField, const MAX_BITS: usize> TableCircuit<E>
         Ok(config.assign_instances(num_witin, num_structural_witin, multiplicity, MAX_BITS)?)
     }
 }
+
+pub struct DoubleRangeTableCircuit<E, const MAX_BITS_1: usize, const MAX_BITS_2: usize, R>(
+    PhantomData<(E, R)>,
+);
+
+impl<E: ExtensionField, const MAX_BITS_1: usize, const MAX_BITS_2: usize, R: RangeTable>
+    TableCircuit<E> for DoubleRangeTableCircuit<E, MAX_BITS_1, MAX_BITS_2, R>
+{
+    type TableConfig = DoubleRangeTableConfig;
+    type FixedInput = ();
+    type WitnessInput = ();
+
+    fn name() -> String {
+        format!("DOUBLE_RANGE_{:?}", R::ROM_TYPE)
+    }
+
+    fn construct_circuit(
+        cb: &mut CircuitBuilder<E>,
+        _params: &ProgramParams,
+    ) -> Result<DoubleRangeTableConfig, ZKVMError> {
```

Comment:
I think we should add assertion here to check `assert_eq!(MAX_BITS_1 + MAX_BITS_2, R::len().ilog2())`